src/generic.jl

```diff
@@ -1239,13 +1239,25 @@ function (\)(A::AbstractMatrix, B::AbstractVecOrMat)
         if istril(A)
             if istriu(A)
                 return Diagonal(A) \ B
+            elseif istriu(A, -1)
+                return Bidiagonal(A, :L) \ B
```

Comment:
Maybe switch the order? Like first test `istriu(A, -1)`, and if that is true, only check `iszero(diagview(A, -1))` to dispatch to `Diagonal(A)`? Otherwise (if I'm not mistaken) you're checking basically the same triangle twice. Same below.